File: backend/src/main/java/org/yougrow/backend/services/QuizService.java
package org.yougrow.backend.services;

import org.yougrow.backend.entities.Quiz;
import org.yougrow.backend.entities.projection.Quiz.QuizDetail;
import org.yougrow.backend.entities.projection.Quiz.QuizShort;

import java.util.List;
import java.util.Optional;

public interface QuizService {

    List<QuizShort> getAllQuizzesShort();

    Optional<QuizDetail> getQuizDetailById(Long id);

    Optional<Quiz> getQuizById(Long id);

    Quiz addQuiz(Quiz quiz);

    Quiz updateQuiz(Quiz quiz);

    void deleteQuiz(Long id);

}

File: frontend/src/app/services/quiz/quiz.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import { QuizDetail } from '../../interfaces/quizDetail';
import { QuizShort } from '../../interfaces/quizShort';

@Injectable({
  providedIn: 'root'
})
export class QuizService {
  URL: string = "https://quizmaster-backend-emku.onrender.com/api/quizzes";

  constructor(private httpClient: HttpClient) {
  }

  public getAllQuizzes(): Observable<QuizShort[]> {
    return this.httpClient.get<QuizShort[]>(`${this.URL}`);
  }

  public getQuizById(id: number): Observable<QuizDetail> {
    return this.httpClient.get<QuizDetail>((`${this.URL}/${id}`));
  }

  public addQuiz(quiz: QuizDetail): Observable<QuizDetail> {
    return this.httpClient.post<QuizDetail>(`${this.URL}`, quiz);
  }

  public updateQuiz(quiz: QuizDetail): Observable<QuizDetail> {
    return this.httpClient.put<QuizDetail>(`${this.URL}`, quiz);
  }

  public deleteQuiz(id: number): Observable<void> {
    return this.httpClient.delete<void>(`${this.URL}/${id}`);
  }

}
